### PR TITLE
[SAR-9168] Added connector service and controller

### PR DESCRIPTION
**Pull request by refaque-ahmad**

### CODE REVIEW

File: app/uk/gov/hmrc/minorentityidentification/controllers/RegisterBusinessEntityController.scala [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class RegisterBusinessEntityController @Inject()(cc: ControllerComponents,
+                                                 registerWithMultipleIdentifiersService: RegisterWithMultipleIdentifiersService,
+                                                 val authConnector: AuthConnector
+                                                )(implicit ec: ExecutionContext) extends BackendController(cc) with AuthorisedFunctions {
+
+  def registerWithSautr(): Action[(Option[String], String)] = Action.async(parse.json[(Option[String], String)](json => for {
+    sautr <- (json \ "trust" \ "sautr").validateOpt[String]
```
Comment (HarrNM) [inline app/uk/gov/hmrc/minorentityidentification/controllers/RegisterBusinessEntityController.scala]: sautr <- (json \ "trust" \ "sautr").validate[String]

File: app/uk/gov/hmrc/minorentityidentification/connectors/RegisterWithMultipleIdentifiersConnector.scala [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          )
+      }
+    }
+  }
+
+  sealed trait RegisterWithMultipleIdentifiersResult
+
+  case class RegisterWithMultipleIdentifiersSuccess(safeId: String) extends RegisterWithMultipleIdentifiersResult
+
+  case class RegisterWithMultipleIdentifiersFailure(status: Int, body: String) extends RegisterWithMultipleIdentifiersResult
```
Comment (HarrNM) [inline app/uk/gov/hmrc/minorentityidentification/connectors/RegisterWithMultipleIdentifiersConnector.scala]: See comment in the controller - if you agree then we won't need this either

File: app/uk/gov/hmrc/minorentityidentification/connectors/RegisterWithMultipleIdentifiersConnector.scala [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  def registerWithSautr(optSautr: Option[String], regime: String
+                      )(implicit hc: HeaderCarrier): Future[RegisterWithMultipleIdentifiersResult] = {
+    val trustIdentifiers = optSautr match {
+      case Some(sautr) => Json.obj("sautr" -> sautr)
+      case _ => Json.obj()
+    }
+
+    val jsonBody: JsObject = Json.obj(
+      "trust" -> trustIdentifiers
+    )
```
Comment (HarrNM) [inline app/uk/gov/hmrc/minorentityidentification/connectors/RegisterWithMultipleIdentifiersConnector.scala]: We would only call this when there is an sautr from the user in the Trust flow so doesn't need to be optional. Not sure what the RegisterAPI would return if we sent a json body with no identifiers

File: app/uk/gov/hmrc/minorentityidentification/controllers/RegisterBusinessEntityController.scala [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          case RegisterWithMultipleIdentifiersSuccess(safeId) =>
+            Ok(Json.obj(
+              "registration" -> Json.obj(
+                "registrationStatus" -> "REGISTERED",
+                "registeredBusinessPartnerId" -> safeId)))
+          case RegisterWithMultipleIdentifiersFailure(status, body) =>
+            Ok(Json.obj(
+              "registration" -> Json.obj(
+                "registrationStatus" -> "REGISTRATION_FAILED")
+            ))
```
Comment (HarrNM) [inline app/uk/gov/hmrc/minorentityidentification/controllers/RegisterBusinessEntityController.scala]: the service doesn't actually return this status. It either returns a RegisterWithMultipleIdentifiersSuccess or throws an Internal Sever Exception so we don't need this bit - not sure why it's in the other services

File: it/uk/gov/hmrc/minorentityidentification/connectors/RegisterWithMultipleIdentifiersConnectorISpec.scala [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import uk.gov.hmrc.minorentityidentification.utils.ComponentSpecHelper
+
+class RegisterWithMultipleIdentifiersConnectorISpec extends ComponentSpecHelper with AuthStub with RegisterWithMultipleIdentifiersStub with FeatureSwitching {
+
+  lazy val connector: RegisterWithMultipleIdentifiersConnector = app.injector.instanceOf[RegisterWithMultipleIdentifiersConnector]
+
+  private implicit val headerCarrier: HeaderCarrier = HeaderCarrier()
+
+  "registerWithMultipleIdentifiers" when {
+    s"the $DesStub feature switch is disabled" when {
```
Comment (HarrNM) [inline it/uk/gov/hmrc/minorentityidentification/connectors/RegisterWithMultipleIdentifiersConnectorISpec.scala]: I think this
s"the $DesStub feature switch is disabled" when { 
should be 
s"the $DesStub feature switch is disabled" should { 
Also maybe add a test for when the call returns a different status (e.g. 500 or 400) and check the InternalServerException is being thrown

File: it/uk/gov/hmrc/minorentityidentification/controllers/RegisterBusinessEntityControllerISpec.scala [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+          "registration" -> Json.obj(
+            "registrationStatus" -> "REGISTERED",
+            "registeredBusinessPartnerId" -> testSafeId))
+
+        val result = post("/register")(jsonBody)
+        result.status mustBe OK
+        result.json mustBe resultJson
+      }
+    }
+    "return INTERNAL_SERVER_ERROR with status Registration_Failed" when {
```
Comment (HarrNM) [inline it/uk/gov/hmrc/minorentityidentification/controllers/RegisterBusinessEntityControllerISpec.scala]: maybe rename to just "return INTERNAL_SERVER_ERROR"

Comment (refaque-ahmad) [reply it/uk/gov/hmrc/minorentityidentification/controllers/RegisterBusinessEntityControllerISpec.scala]: done

File: app/uk/gov/hmrc/minorentityidentification/connectors/RegisterWithMultipleIdentifiersConnector.scala (hunk 3 above)
Comment (refaque-ahmad) [reply app/uk/gov/hmrc/minorentityidentification/connectors/RegisterWithMultipleIdentifiersConnector.scala]: done

File: app/uk/gov/hmrc/minorentityidentification/connectors/RegisterWithMultipleIdentifiersConnector.scala (hunk 2 above)
Comment (refaque-ahmad) [reply app/uk/gov/hmrc/minorentityidentification/connectors/RegisterWithMultipleIdentifiersConnector.scala]: done

File: app/uk/gov/hmrc/minorentityidentification/controllers/RegisterBusinessEntityController.scala (hunk 1 above)
Comment (refaque-ahmad) [reply app/uk/gov/hmrc/minorentityidentification/controllers/RegisterBusinessEntityController.scala]: done

File: app/uk/gov/hmrc/minorentityidentification/controllers/RegisterBusinessEntityController.scala (hunk 4 above)
Comment (refaque-ahmad) [reply app/uk/gov/hmrc/minorentityidentification/controllers/RegisterBusinessEntityController.scala]: done

File: it/uk/gov/hmrc/minorentityidentification/connectors/RegisterWithMultipleIdentifiersConnectorISpec.scala (hunk 5 above)
Comment (refaque-ahmad) [reply it/uk/gov/hmrc/minorentityidentification/connectors/RegisterWithMultipleIdentifiersConnectorISpec.scala]: updated and added

**Review by HarrNM — APPROVED** [review]

**Pull request merged**